### PR TITLE
feat(next): done - add title to pages

### PR DESCRIPTION
**Pull request by chrisdadev13**
## What does this PR do?

Hi everyone! This PR adds document titles to each page indicating information about the page/convo. I started this task in the NextJS workspace, but when I saw that the issue was specified for Nuxt/Vue I decided to add the change here too, I'm not as good at Nuxt as I am at Next (even though the change is pretty straight forward) I would appreciate your feedback 🙏 (this is my first pr ever too)
 
Fixes (https://github.com/un/inbox/issues/215)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by McPizza0** [comment]
Hey @chrisdadev13 
Sorry, this issue should have been re-labeled
we've stopped all work on the NuxtJS version and now only make changes to NextJS (in the `/apps/web` folder)
do you want to make the code changes there since you're more familiar with it?

**Comment by chrisdadev13** [comment]
> Hey @chrisdadev13 Sorry, this issue should have been re-labeled we've stopped all work on the NuxtJS version and now only make changes to NextJS (in the `/apps/web` folder) do you want to make the code changes there since you're more familiar with it?

Sure thing! I already have the changes there 🙂 Just have to prepare it

File: apps/web/src/app/layout.tsx
Comment (chrisdadev13) [inline apps/web/src/app/layout.tsx:1]: There was a bug opening convos because the "TooltipProvider" wasn't present

**Comment by McPizza0** [comment]
Thats an interesting approach @chrisdadev13 
this way would inject the metadata into the body of the page, rather than the head - i think theres implications for that

we're generally using `use-client` for all pages, only using SSR for login/signup
does the standard `generateMetadata` not work with normal clients: https://nextjs.org/docs/app/api-reference/functions/generate-metadata#generatemetadata-function

**Comment by chrisdadev13** [comment]
> Thats an interesting approach @chrisdadev13 this way would inject the metadata into the body of the page, rather than the head - i think theres implications for that
> 
> we're generally using `use-client` for all pages, only using SSR for login/signup does the standard `generateMetadata` not work with normal clients: https://nextjs.org/docs/app/api-reference/functions/generate-metadata#generatemetadata-function

Yes, that is true, and sadly at this time the `generateMetadata` only works with **server components**.  That is why I came up with that (Metadata Component) solution.
Screenshot 2024-07-04 at 9 32 37 AM

That example I left in the comment, is how they actually recommend adding `metadata` on client pages. I'll try to split the PR and do it like that for few pages, if it's okay 👍 @McPizza0

File: apps/web/src/app/layout.tsx
Comment (BlankParticle) [reply apps/web/src/app/layout.tsx:1]: I fixed this issue in a recent PR, this issue was introduced in a previous PR. The fixed PR is not merged yet.
You can fix the issue in your PR for now. I will rebase this PR before merging

**Comment by chrisdadev13** [comment]
I'll close this one to make the changes more focused, here is the PR to add metadata dynamically on convo routes and statically on the page to create a new convo https://github.com/un/inbox/pull/542

**Pull request closed without merging**